File: algorithm-problems/src/main/java/us/supercheng/algorithm/problems/designanorderedstream/OrderedStream.js
/**
 * @param {number} n
 */
var OrderedStream = function(n) {
    this.ptr = 1;
    this.map = {};
};

/**
 * @param {number} id
 * @param {string} value
 * @return {string[]}
 */
OrderedStream.prototype.insert = function(id, value) {
    var ret = [];
    this.map[id] = value;

    for (; this.map[this.ptr]; this.ptr += 1)
        ret.push(this.map[this.ptr]);

    return ret;
};

File: algorithm-problems/src/main/java/us/supercheng/algorithm/problems/designanorderedstream/OrderedStream.java
package us.supercheng.algorithm.problems.designanorderedstream;

import java.util.ArrayList;
import java.util.List;

public class OrderedStream {

    private String[] arr;
    private int prt;
    private int L;

    public OrderedStream(int n) {
        this.prt = 1;
        this.L = n + 1;
        this.arr = new String[this.L];
    }

    public List<String> insert(int id, String value) {
        List<String> ret = new ArrayList<>();
        this.arr[id] = value;

        while (this.prt < this.L && this.arr[this.prt] != null)
            ret.add(this.arr[this.prt++]);

        return ret;
    }
}